vowpalwabbit/gd.cc

```diff
@@ -496,7 +496,10 @@ inline void pred_per_update_feature(norm_data& nd, float x, float& fw)
       x2 = x2_min;
     }
     if (x2 > x2_max)
-      THROW("your features have too much magnitude");
+      {
+        x2 = x2_max;
```

Comment:
x2_max is FLT_MAX which is relatively easily can be converted into inf (i.e. by multiplying by 2). So it seems like now we have good chances to propagate inf into weights (i.e. line 511) which seems problematic.

Reply:
it was always the case for just a little bit smaller numbers though.